File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/templatemethod/transport/Train.java
// Train.java

package com.bigboxcode.designpattern.templatemethod.transport;

public class Train extends Transport {
    @Override
    void createBody() {
        System.out.println("Creating Train Body");
    }

    @Override
    void addEngine() {
        System.out.println("Adding Engine to Train");
    }

    @Override
    void addWheel() {
        System.out.println("Adding Wheels to Train");
    }

    @Override
    void addWing() {
        // Not required for Train
    }
}

File: design-patterns/ts/template-method/transport/train.ts
// train.ts

import Transport from "./transport";

class Train extends Transport {
    createBody(): void {
        console.log("Creating Train Body");
    }

    addEngine(): void {
        console.log("Adding Engine to Train");
    }

    addWheel(): void {
        console.log("Adding Wheels to Train");
    }

    addWing() : void{
        // not required for train
    }
}

export default Train;